bfvmm/src/hve/arch/intel_x64/ept_violation.cpp

```diff
@@ -0,0 +1,222 @@
+//
+// Bareflank Extended APIs
+// Copyright (C) 2018 Assured Information Security, Inc.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#include <bfdebug.h>
+#include <hve/arch/intel_x64/hve.h>
+
+namespace eapis
+{
+namespace intel_x64
+{
+
+static bool
+default_handler(
+    gsl::not_null<vmcs_t *> vmcs, ept_violation::info_t &info)
+{ bfignored(vmcs); bfignored(info); return true; }
+
+ept_violation::ept_violation(
+    gsl::not_null<eapis::intel_x64::hve *> hve
+) :
+    m_exit_handler{hve->exit_handler()}
+{
+    using namespace vmcs_n;
+
+    m_exit_handler->add_handler(
+        exit_reason::basic_exit_reason::ept_violation,
+        ::handler_delegate_t::create<ept_violation, &ept_violation::handle>(this)
+    );
+
+    this->add_read_handler(handler_delegate_t::create<default_handler>());
```

Comment:
same thing, no default handlers should be provided. 

As another note, your integration tests should be testing all of this logic. Another way to look at it is, if you cannot write an integration test that exercises the default handler, you probably should not have one.